Add `ddp_*_find_unused_parameters_false` to Plugins Registry. (#8483)
PyTorchLightning/pytorch-lightning: docs/source/guides/speed.rst

Whereas :class:`~pytorch_lightning.plugins.training_type.DDPPlugin` only performs 1 transfer to sync gradients, making DDP MUCH faster than DP.

> When using DDP plugins, set find_unused_parameters=False

Reviewer: ```suggestion
When using DDP plugins, set ``find_unused_parameters=False``
```
maybe?

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> By default we have set ``find_unused_parameters`` to True for compatibility issues that have arisen in the past (see the `discussion <https://github.com/PyTorchLightning/pytorch-lightning/discussions/6219>`_ for more information).

Reviewer: ```suggestion
By default we have set ``find_unused_parameters`` to True for compatibility reasons that have been observed in the past (see the `discussion <https://github.com/PyTorchLightning/pytorch-lightning/discussions/6219>`_ for more details).
```

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer 2: Please undo the ticks or otherwise the title will not display correctly: 

![image](https://user-images.githubusercontent.com/5495193/126626976-cb76787a-8bce-443a-b93d-6ba2c6160362.png)

Reviewer 2: ```suggestion
When using DDP plugins, set find_unused_parameters=False
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

> When using DDP plugins, set ``find_unused_parameters=False``
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer 2: ```suggestion
When using DDP plugins, set find_unused_parameters=False
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

> This by default comes with a performance hit, and can be disabled in most cases.

Reviewer: any info how big is this hit, in %?

Author: I don't get what you mean?

Reviewer: > comes with a performance hit

so how big is the hit?

Author: >traversing the autograd graph introduces extra overheads, so applications should only set find_unused_parameters to True when necessary.

> .. note::

Reviewer 3: We always use notes :D

```suggestion
.. tip::
```

Whereas :class:`~pytorch_lightning.plugins.training_type.DDPPlugin` only performs 1 transfer to sync gradients, making DDP MUCH faster than DP.

> When using DDP or DDPSpawn, set find_unused_parameters=False

Reviewer 2: the term DDP already covered the family of ddp plugins. spawn is not the only one, there are others. 
so I think a simple note that this applies to all plugins supporting the `find_unused_parameters` would suffice

Author: Updated!

Reviewer 2: love it

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer 3: Shouldn't this section also mention the availability of the `str`? Or do we mention it somewhere else?

Author: `PluginsRegistry` will have a separate section with multiple DeepSpeed aliases and this.

        plugins=DDPPlugin(find_unused_parameters=False),
    )

> .. code-block::

Reviewer: ```suggestion
.. testcode::
```

> .. code-block:: python

Reviewer: ```suggestion
.. code-block::
```
the after `::` is a label not language type

    from pytorch_lightning.plugins import DDPSpawnPlugin

    trainer = pl.Trainer(
        gpus=2,
        plugins=DDPSpawnPlugin(find_unused_parameters=False),

>     )

Reviewer: this can be `testcode` rather, right?

Author: An example instead, we have internal tests.

Reviewer: yes but rather keep all examples aligned with api

Author: `PluginsRegistry` will have a separate section with multiple DeepSpeed aliases and this. Will add it.
